### PR TITLE
Another branch

### PR DESCRIPTION
**Pull request by charz**
Test to another.

### CODE REVIEW

**Comment by charz** [comment]
test1

**Comment by petertom51** [comment]
Test 2

**Comment by charz** [comment]
ok, I'll do few things.

### git pull origin master

**Comment by petertom51** [comment]
## git pull origin master

Does this lead a merge point?

**Comment by charz** [comment]
```
[22:14][][charles@CharzMac: ~/github/test_rebase (another_branch)] $ git remote add petertom https://github.com/petertom51/test_rebase.git
[22:14][][charles@CharzMac: ~/github/test_rebase (another_branch)] $ git remote show
origin
petertom
[22:14][][charles@CharzMac: ~/github/test_rebase (another_branch)] $ git pull petertom master
remote: Counting objects: 6, done.
remote: Compressing objects: 100% (5/5), done.
remote: Total 6 (delta 1), reused 6 (delta 1)
Unpacking objects: 100% (6/6), done.
From https://github.com/petertom51/test_rebase
 * branch            master     -> FETCH_HEAD
 * [new branch]      master     -> petertom/master
Auto-merging Test_File
CONFLICT (content): Merge conflict in Test_File
Auto-merging README.md
CONFLICT (content): Merge conflict in README.md
Automatic merge failed; fix conflicts and then commit the result.
[22:14][][charles@CharzMac: ~/github/test_rebase (another_branch)] $ 
```

**Comment by charz** [comment]
```
[22:14][][charles@CharzMac: ~/github/test_rebase (another_branch)] $ git status
On branch another_branch
You have unmerged paths.
  (fix conflicts and run "git commit")

Unmerged paths:
  (use "git add <file>..." to mark resolution)

	both modified:   README.md
	both modified:   Test_File

Untracked files:
  (use "git add <file>..." to include in what will be committed)

	test

no changes added to commit (use "git add" and/or "git commit -a")
[22:15][][charles@CharzMac: ~/github/test_rebase (another_branch)] $ vim README.md 
[22:15][][charles@CharzMac: ~/github/test_rebase (another_branch)] $ git status
On branch another_branch
You have unmerged paths.
  (fix conflicts and run "git commit")

Unmerged paths:
  (use "git add <file>..." to mark resolution)

	both modified:   README.md
	both modified:   Test_File

Untracked files:
  (use "git add <file>..." to include in what will be committed)

	test

no changes added to commit (use "git add" and/or "git commit -a")
[22:15][][charles@CharzMac: ~/github/test_rebase (another_branch)] $ git Test_File 
git: 'Test_File' is not a git command. See 'git --help'.
[22:15][][charles@CharzMac: ~/github/test_rebase (another_branch)] $ vim Test_File 
[22:15][][charles@CharzMac: ~/github/test_rebase (another_branch)] $ git status
On branch another_branch
You have unmerged paths.
  (fix conflicts and run "git commit")

Unmerged paths:
  (use "git add <file>..." to mark resolution)

	both modified:   README.md
	both modified:   Test_File

Untracked files:
  (use "git add <file>..." to include in what will be committed)

	test

no changes added to commit (use "git add" and/or "git commit -a")
[22:15][][charles@CharzMac: ~/github/test_rebase (another_branch)] $ git add README.md Test_File 
[22:16][][charles@CharzMac: ~/github/test_rebase (another_branch)] $ git commit
[another_branch 03c5ad2] Merge branch 'master' of https://github.com/petertom51/test_rebase into another_branch
```

**Comment by charz** [comment]
image

**Comment by charz** [comment]
image

**Comment by petertom51** [comment]
YEAH, you got a merge point

**Comment by charz** [comment]
Reset it again, here is a right way.

```
[22:18][][charles@CharzMac: ~/github/test_rebase (another_branch)] $ git checkout master
Switched to branch 'master'
Your branch is up-to-date with 'origin/master'.
[22:19][][charles@CharzMac: ~/github/test_rebase (master)] $ git pull
Already up-to-date.
[22:19][][charles@CharzMac: ~/github/test_rebase (master)] $ git pull petertom master
From https://github.com/petertom51/test_rebase
 * branch            master     -> FETCH_HEAD
Updating aa8e7c8..a19c932
Fast-forward
 README.md | 2 ++
 Test_File | 2 ++
 2 files changed, 4 insertions(+)
[22:19][][charles@CharzMac: ~/github/test_rebase (master)] $ git checkout another 
error: pathspec 'another' did not match any file(s) known to git.
[22:19][][charles@CharzMac: ~/github/test_rebase (master)] $ git checkout another_branch
Switched to branch 'another_branch'
[22:19][][charles@CharzMac: ~/github/test_rebase (another_branch)] $ git rebase master
First, rewinding head to replay your work on top of it...
Applying: Change Test_file.
Using index info to reconstruct a base tree...
M   Test_File
Falling back to patching base and 3-way merge...
Auto-merging Test_File
CONFLICT (content): Merge conflict in Test_File
Failed to merge in the changes.
Patch failed at 0001 Change Test_file.
The copy of the patch that failed is found in:
   /Users/charles/github/test_rebase/.git/rebase-apply/patch

When you have resolved this problem, run "git rebase --continue".
If you prefer to skip this patch, run "git rebase --skip" instead.
To check out the original branch and stop rebasing, run "git rebase --abort".

[22:19][][charles@CharzMac: ~/github/test_rebase ((no branch, rebasing another_branch))] $ git status
rebase in progress; onto a19c932
You are currently rebasing branch 'another_branch' on 'a19c932'.
  (fix conflicts and then run "git rebase --continue")
  (use "git rebase --skip" to skip this patch)
  (use "git rebase --abort" to check out the original branch)

Unmerged paths:
  (use "git reset HEAD <file>..." to unstage)
  (use "git add <file>..." to mark resolution)

    both modified:   Test_File

Untracked files:
  (use "git add <file>..." to include in what will be committed)

    test

no changes added to commit (use "git add" and/or "git commit -a")
[22:19][][charles@CharzMac: ~/github/test_rebase ((no branch, rebasing another_branch))] $ vim Test_File 
[22:20][][charles@CharzMac: ~/github/test_rebase ((no branch, rebasing another_branch))] $ git add Test_File 
[22:20][][charles@CharzMac: ~/github/test_rebase ((no branch, rebasing another_branch))] $ git status
rebase in progress; onto a19c932
You are currently rebasing branch 'another_branch' on 'a19c932'.
  (all conflicts fixed: run "git rebase --continue")

Changes to be committed:
  (use "git reset HEAD <file>..." to unstage)

    modified:   Test_File

Untracked files:
  (use "git add <file>..." to include in what will be committed)

    test

[22:20][][charles@CharzMac: ~/github/test_rebase ((no branch, rebasing another_branch))] $ git rebase --continue
Applying: Change Test_file.
Applying: Add item1, item2.
Using index info to reconstruct a base tree...
M   README.md
Falling back to patching base and 3-way merge...
Auto-merging README.md
CONFLICT (content): Merge conflict in README.md
Failed to merge in the changes.
Patch failed at 0002 Add item1, item2.
The copy of the patch that failed is found in:
   /Users/charles/github/test_rebase/.git/rebase-apply/patch

When you have resolved this problem, run "git rebase --continue".
If you prefer to skip this patch, run "git rebase --skip" instead.
To check out the original branch and stop rebasing, run "git rebase --abort".

[22:20][][charles@CharzMac: ~/github/test_rebase ((no branch, rebasing another_branch))] $ git status
rebase in progress; onto a19c932
You are currently rebasing branch 'another_branch' on 'a19c932'.
  (fix conflicts and then run "git rebase --continue")
  (use "git rebase --skip" to skip this patch)
  (use "git rebase --abort" to check out the original branch)

Unmerged paths:
  (use "git reset HEAD <file>..." to unstage)
  (use "git add <file>..." to mark resolution)

    both modified:   README.md

Untracked files:
  (use "git add <file>..." to include in what will be committed)

    test

no changes added to commit (use "git add" and/or "git commit -a")
[22:20][][charles@CharzMac: ~/github/test_rebase ((no branch, rebasing another_branch))] $ vim README.md 
[22:21][][charles@CharzMac: ~/github/test_rebase ((no branch, rebasing another_branch))] $ git status
rebase in progress; onto a19c932
You are currently rebasing branch 'another_branch' on 'a19c932'.
  (fix conflicts and then run "git rebase --continue")
  (use "git rebase --skip" to skip this patch)
  (use "git rebase --abort" to check out the original branch)

Unmerged paths:
  (use "git reset HEAD <file>..." to unstage)
  (use "git add <file>..." to mark resolution)

    both modified:   README.md

Untracked files:
  (use "git add <file>..." to include in what will be committed)

    test

no changes added to commit (use "git add" and/or "git commit -a")
[22:21][][charles@CharzMac: ~/github/test_rebase ((no branch, rebasing another_branch))] $ git add README.md 
[22:21][][charles@CharzMac: ~/github/test_rebase ((no branch, rebasing another_branch))] $ git status
rebase in progress; onto a19c932
You are currently rebasing branch 'another_branch' on 'a19c932'.
  (all conflicts fixed: run "git rebase --continue")

Changes to be committed:
  (use "git reset HEAD <file>..." to unstage)

    modified:   README.md

Untracked files:
  (use "git add <file>..." to include in what will be committed)

    test

[22:21][][charles@CharzMac: ~/github/test_rebase ((no branch, rebasing another_branch))] $ git rebase --continue
Applying: Add item1, item2.
```

**Comment by petertom51** [comment]
some comments disappear.

**Comment by charz** [comment]
I deleted them, no worry! :)

**Comment by petertom51** [comment]
XD

**Comment by charz** [comment]
Here is now screenshot of my local git.
image

**Comment by charz** [comment]
Push code to `another_branch`

```
[22:23][][charles@CharzMac: ~/github/test_rebase (another_branch)] $ git push origin another_branch
Username for 'https://github.com': charz
Password for 'https://charz@github.com': 
To https://github.com/charz/test_rebase
 ! [rejected]        another_branch -> another_branch (non-fast-forward)
error: failed to push some refs to 'https://github.com/charz/test_rebase'
hint: Updates were rejected because the tip of your current branch is behind
hint: its remote counterpart. Integrate the remote changes (e.g.
hint: 'git pull ...') before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.
```

It works with `--force` option.

```
[22:24][][charles@CharzMac: ~/github/test_rebase (another_branch)] $ git push origin another_branch --force
Username for 'https://github.com': charz
Password for 'https://charz@github.com': 
Counting objects: 12, done.
Delta compression using up to 4 threads.
Compressing objects: 100% (12/12), done.
Writing objects: 100% (12/12), 1.17 KiB | 0 bytes/s, done.
Total 12 (delta 3), reused 0 (delta 0)
To https://github.com/charz/test_rebase
 + ec9875d...9eb49f8 another_branch -> another_branch (forced update)
```

**Comment by petertom51** [comment]
amazing...

**Comment by charz** [comment]
Do you see the position of commits were change?

File: README.md [hunk 1]
```diff
@@ -1,3 +1,6 @@
 # test_rebase
 
 ```this file to show description```
+### item1
```
Comment (petertom51) [inline README.md:4]: will I do some comment that the conversation will disapeear?

File: Test_File [hunk 2]
```diff
@@ -3,3 +3,7 @@ This is a first file to commit.
 This repository is just created for test rebase and the conversation should not disapear.
 
 Try to modify this file.
+Add message by charles.
+
```
Comment (petertom51) [inline Test_File:7]: And also here?

File: README.md (hunk 1 above)
Comment (petertom51) [inline README.md:4]: after rebase again..

**Comment by charz** [comment]
Summary:
- Add remote repo in local git (`git remote add <remote_name> <git_url>`)
- Switch to master branch and pull remote branch (`git pull petertom master`)
- Switch to local_branch that you created and do rebase (`git rebase master`)
- Fixed CONFLICT files and do `git rebase --continue` until it done.
- Push to you remote repo

File: Test_File (hunk 2 above)
Comment (charz) [inline Test_File:7]: Yeah

**Pull request merged**